teps/0100-embedded-taskruns-and-runs-status-in-pipelineruns.md

```diff
@@ -171,12 +201,503 @@ In addition to grouping related resources, Tekton Results provides long term sto
+
+Taking the [example](#example) above, this would be the status of the `TaskRun`:
+
+```yaml
+apiVersion: tekton.dev/v1beta1
+kind: TaskRun
+metadata:
+  labels:
+    app.kubernetes.io/managed-by: tekton-pipelines
+    tekton.dev/memberOf: tasks
+    tekton.dev/pipeline: triggers-release-nightly
+    tekton.dev/pipelineRun: triggers-release-nightly
+    tekton.dev/pipelineTask: build
+    tekton.dev/task: build
+  name: triggers-release-nightly-build
+  ownerReferences:
+    - apiVersion: tekton.dev/v1beta1
+      blockOwnerDeletion: true
+      controller: true
+      kind: PipelineRun
+      name: triggers-release-nightly
+spec:
+    ...
+status:
+    completionTime: "2020-05-04T02:10:49Z"
+    conditions:
+      - lastTransitionTime: "2020-05-04T02:10:49Z"
+        message: All Steps have completed executing
+        reason: Succeeded
+        status: "True"
+        type: Succeeded
+    podName: triggers-release-nightly-build-pod
+    resourcesResult:
+      - key: commit
+        resourceRef:
+          name: git-source-triggers
+        value: 9ab5a1234166a89db352afa28f499d596ebb48db
+    startTime: "2020-05-04T02:05:07Z"
+    steps:
+      - container: step-build
+        imageID: docker-pullable://golang@sha256:a90f267133
+        name: build
+        terminated:
+          containerID: docker://6b6471f501f59dbb
+          exitCode: 0
+          finishedAt: "2020-05-04T02:10:45Z"
+          reason: Completed
+          startedAt: "2020-05-04T02:06:24Z"
+```
+
+#### Results API
+
+If the cluster has been cleaned up, a user can rely on the Results API to get the full
+details of the `PipelineRun`'s `TaskRuns` and `Runs`. They can use the Tekton CLI plugin
+for the Tekton Results API:
+
+```shell
+tkn-results records list default/results/- --filter name=="triggers-release-nightly"
+```
+
+For more details on fetching `Results` and `Records`, see the [documentation][cli-plugin].
+
+#### Go Libraries
+
+We will provide functions in the Go libraries to get the `TaskRuns` and `Runs` of a
+given `PipelineRun`. These functions will be useful for the Tekton Dashboard, Tekton
+CLI and other projects using our Go libraries.
+
+### Beta API
+
+Because the `PipelineRun` status is part of the [Pipelines API][api-definition],
+[removing the full embedded statuses](#2-deprecate-and-remove-full-embedded-status)
+is backwards incompatible. 
+
+To support migration as required by our [API compatibility policy][deprecations]
+we will add a behavior flag - `embedded-status` - used to configure whether the
+`PipelineRuns` should contain:
+* the full embedded status of its `TaskRuns` and `Runs`.
+* the minimal embedded status of its `TaskRuns` and `Runs`.
+* both the full and minimal status of its `TaskRuns` and `Runs` - this provides a
+  smoother transition for users and tools.
+
+Following our [policy][behavior-flags] on updating behavior flags:
+1. The `embedded-status` flag will be `full` by default, users can set it to `minimal`
+   or `both`.
```

Comment:
Thank you @jerop for a thorough transition plan but I think we can simplify it a little. 

I don't see the value in supporting `full` option and making it `default`. Having default as `full` implies no changes in the existing behavior so why would I change my controller configuration?

Also, I don't see value in `both`, it will even make the status more verbose. 

I agree with @vdemeester's comments [here](https://github.com/tektoncd/community/pull/616#discussion_r798732028). 

If I want to take advantage of this new functionality, I will just set the `embedded-status` to `minimal`.

And after 9 months making sure the CLI/Dashboard have adopted this new functionality, we can remove this flag completely and make the `minimal` status default.

What do you think?

Reply:
Thanks for the review @pritidesai - updated the TEP as we discussed in the API working group meeting yesterday :)